service/src/java/org/apache/hive/service/cli/operation/OperationManager.java

```diff
@@ -94,6 +95,14 @@ public synchronized void start() {
   @Override
   public synchronized void stop() {
     super.stop();
+    for (Operation operation : getOperations()) {
+      try {
```

Comment:
Trying to understand how this change is related to decommissioning logic? Before this patch, how are we cancelling operations when hs2.stop is called?

Reply:
we actually do not cancel operations at all before the changes, instead we shutdown the backgroud pool running the operations: https://github.com/apache/hive/blob/master/service/src/java/org/apache/hive/service/cli/session/SessionManager.java#L367
this does not guarantee that the HiveServer2 would stop immediately when hs2.stop is finished. In some cases, it would wait for all operations done before exiting, and some operation's local files may not be cleaned.

Reply:
The CliService stops its service on hs2.stop() firstly before OperationManager calling stop(), so it's safe to cancel operations here.

Reply:
Thanks for the explanation.